File: backend/app/models/Penjualan.go
package models

import "time"

type Penjualan struct {
	ID                   string     `json:"id"`
	CabangID             string     `json:"cabang_id"`
	ShiftKasirID         *string    `json:"shift_kasir_id"`
	PelangganID          *string    `json:"pelanggan_id"`
	NomorInvoice         string     `json:"nomor_invoice"`
	Tanggal              time.Time  `json:"tanggal"`
	TipePenjualan        string     `json:"tipe_penjualan"`
	Subtotal             float64    `json:"subtotal"`
	DiskonTotal          float64    `json:"diskon_total"`
	PajakTotal           float64    `json:"pajak_total"`
	GrandTotal           float64    `json:"grand_total"`
	BayarTotal           float64    `json:"bayar_total"`
	Kembalian            float64    `json:"kembalian"`
	StatusBayar          string     `json:"status_bayar"`
	Status               string     `json:"status"`
	Catatan              *string    `json:"catatan"`
	DibuatOleh           *string    `json:"dibuat_oleh"`
	PaymentProvider      *string    `json:"payment_provider"`
	PaymentExternalID    *string    `json:"payment_external_id"`
	PaymentProviderID    *string    `json:"payment_provider_id"`
	PaymentTransactionID *string    `json:"payment_transaction_id"`
	PaymentURL           *string    `json:"payment_url"`
	PaymentStatus        *string    `json:"payment_status"`
	PaymentExpiresAt     *time.Time `json:"payment_expires_at"`
	AccurateInvoiceID    *int64     `json:"accurate_invoice_id"`
	AccurateSyncStatus   *string    `json:"accurate_sync_status"`
	AccurateSyncError    *string    `json:"accurate_sync_error"`
	CreatedAt            time.Time  `json:"created_at"`
	UpdatedAt            time.Time  `json:"updated_at"`
}

File: src/types/penjualan.ts
export type MetodePembayaran = "tunai" | "transfer" | "accurate";
export type StatusPenjualan =
  | "menunggu_pembayaran"
  | "selesai"
  | "gagal"
  | "dibatalkan";
export type StatusBayar = "belum_bayar" | "lunas" | "sebagian";

export interface PenjualanDetail {
  id: string;
  penjualanId: string;
  barangId: string;
  namaBarang: string;
  batchId?: string;
  satuanId?: string;
  satuanNama?: string;
  jumlah: number;
  hargaJual: number;
  diskonPersen: number;
  diskonNominal: number;
  subtotal: number;
  hargaPokok: number;
}

export interface Penjualan {
  id: string;
  cabangId: string;
  shiftKasirId?: string;
  nomorInvoice: string;
  pelangganId?: string;
  namaPelanggan: string;
  resepId?: string;
  tanggal: string;
  tipePenjualan: string;
  subtotal: number;
  diskonTotal: number;
  pajakTotal: number;
  grandTotal: number;
  bayarTotal: number;
  kembalian: number;
  statusBayar: StatusBayar;
  status: StatusPenjualan;
  metodePembayaran?: MetodePembayaran;
  catatan?: string;
  createdBy?: string;
  createdAt: string;
  details: PenjualanDetail[];
}

export interface CartItem {
  cartKey?: string;
  barangId: string;
  kode: string;
  nama: string;
  satuanId?: string;
  satuanNama?: string;
  tipeHarga?: "jual" | "eceran";
  stockQtyPerUnit?: number;
  hargaJual: number;
  stokTersedia: number;
  membutuhkanResep: boolean;
  quantity: number;
}
